mdn/kuma
# Subscribe user to newsletter on save

Fixes #5792 

## What it does
It override the `User`-model's save method to check whether the `salesforce_connection` "enum" is set to `pending` and whether it was empty or `error` before. If those are true it subscribes that user to our newsletter on Salesforce, through [basket](https://basket.readthedocs.io/newsletter_api.html#news-subscribe).

## How to test it

To test this PR, either run the user edit/signup request with curl/Postman/... and add `user-salesforce_connection: 'pending'` to the params or temporarily apply this patch:

<details>
  <summary>user_edit.html patch</summary>

```Index: kuma/users/jinja2/users/user_edit.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- kuma/users/jinja2/users/user_edit.html	(revision aa879155482cd67e8b6aee566b8aa7b39dbd9010)
+++ kuma/users/jinja2/users/user_edit.html	(date 1574787500681)
@@ -121,6 +121,13 @@
             {% endif %}
           </li>
 
+            <li class="field">
+                <label for="{{ user_form.salesforce_connection.id_for_label }}">
+                    {{ user_form.salesforce_connection }}
+                    {{ user_form.salesforce_connection.label }}
+                </label>
+            </li>
+
           {{ li_field(user_form, 'twitter_url', classes='site twitter') }}
 
           {{ li_field(user_form, 'stackoverflow_url', classes='site stackoverflow') }}
```
</details>

Now after setting the flag, you should receive an email, welcoming you to the MDN newsletter.
